pkg/nsx/services/inventory/build.go

```diff
@@ -288,7 +288,20 @@ func (s *InventoryService) BuildService(service *corev1.Service) (retry bool) {
 		netStatus = NetworkStatusUnhealthy
 	}
 
-	// Update the Pods' service IDs which are related to this service
+	// Get network errors from service annotations
+	var networkErrors []common.NetworkError
+	if status != InventoryStatusUp {
+		// Check for NCP errors in service annotations
+		for key, value := range service.Annotations {
+			if util.Contains([]string{NcpLbError, NcpLbPortError, NcpLbEpError, NcpDlbError, NcpSnatError}, key) {
```

Comment:
Maybe define something like serviceNCPErrors := []string{NcpLbError, NcpLbPortError, NcpLbEpError, NcpDlbError, NcpSnatError} for reusing? It could be out of loop or even global.